content/guides/testing-strategies/working-with-graphql.md

```diff
@@ -0,0 +1,159 @@
+
+```js
+// app.spec.js
+import {
+  hasOperationName,
+  aliasQuery,
+  aliasMutation,
+} from '../utils/graphql-test-utils'
+
+context('Tests', () => {
+  beforeEach(() => {
+    cy.intercept('POST', apiGraphQL, (req) => {
+      // Queries
+      aliasQuery(req, 'GetLaunchList')
+      aliasQuery(req, 'LaunchDetails')
+      aliasQuery(req, 'GetMyTrips')
+
+      // Mutations
+      aliasMutation(req, 'Login')
+      aliasMutation(req, 'BookTrips')
+    })
+  })
+  // ...
+})
+```
+
+## Expectations for Query or Mutation Results
+
+Expectations can be made against the response of an intercepted GraphQL query or mutation using [cy.wait()](/api/commands/wait).
+
+```js
+// app.spec.js
+import { hasOperationName, aliasQuery } from '../utils/graphql-test-utils'
+
+context('Tests', () => {
+  beforeEach(() => {
+    cy.intercept('POST', apiGraphQL, (req) => {
+      // Queries
+      aliasQuery(req, 'Login')
+
+      // ...
+    })
+  })
+
+  it('should verify login data', () => {
+    cy.wait('@gqlLoginQuery').then((resp) => {
+      expect(resp.response.body.data.login.id).to.exist
+      expect(resp.response.body.data.login.token).to.exist
+    })
+  })
+})
+```
+
+## Modifying a Query or Mutation Response
+
+As of version 7.0 of Cypress, [cy.intercept()](/api/commands/intercept) can override the response to a GraphQL query or mutation by declaring an intercept in the test.
+
+In the test below, the response is modified to test the UI for a single page of results.
+
+```js
+// app.spec.js
+import { hasOperationName, aliasQuery } from '../utils/graphql-test-utils'
+
+context('Tests', () => {
+  beforeEach(() => {
+    cy.intercept('POST', apiGraphQL, (req) => {
+      // Queries
+      aliasQuery(req, 'GetLaunchList')
+
+      // ...
+    })
+  })
+
+  it('should not display the load more button on the launches page', () => {
+    cy.intercept('POST', apiGraphQL, (req) => {
+      const { body } = req
+      if (hasOperationName(req, 'GetLaunchList')) {
+        // Declare the alias from the initial intercept in the beforeEach
+        req.alias = 'gqlGetLaunchListQuery'
+
+        // Set req.fixture or use req.reply to modify portions of the response
+        req.reply((res) => {
+          // Modify the response body directly
+          res.body.data.launches.hasMore = false
+          res.body.data.launches.launches = res.body.data.launches.launches.slice(
+            5
+          )
+        })
+      }
+    })
+
+    // Must visit after cy.intercept
+    cy.visit('/')
+
+    cy.wait('@gqlGetLaunchListQuery').then((resp) => {
+      expect(resp.response.body.data.launches.hasMore).to.be.false
+      expect(resp.response.body.data.launches.launches.length).to.be.lte(20)
+    })
+
+    cy.getBySelLike('launch-list-tile').its('length').should('be.gte', 1)
+    cy.getBySelLike('launch-list-tile').its('length').should('be.lt', 20)
+
+    cy.getBySelLike('launches-load-more-button').should('not.exist')
```

Comment:
Users that have no experience with RWA may not understand that `getBySelLike` is not a native command. 
```suggestion
    cy.get('#launch-list')
       .its('length')
       .should('be.gte', 1)
       .and('be.lt', 20)
    cy.contains('button', 'Load More')
       .should('not.exist')
```